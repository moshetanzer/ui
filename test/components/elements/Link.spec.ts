import { describe, it, expect } from 'vitest'
import { ULink } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Link', () => {
  it.each([
    ['with as', { props: { as: 'div' } }],
    ['with to', { props: { to: '/' } }],
    ['with type', { props: { type: 'submit' } }],
    ['with disabled', { props: { disabled: true } }],
    ['with activeClass', { props: { active: true, to: '/', activeClass: 'text-sm' } }],
    ['with inactiveClass', { props: { active: false, to: '/', inactiveClass: 'text-gray-300' } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof ULink.props>) => {
    const html = await ComponentRender(nameOrHtml, options, ULink)
    expect(html).toMatchSnapshot()
  })
})
