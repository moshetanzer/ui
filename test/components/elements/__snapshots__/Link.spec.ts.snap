// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Link > renders with activeClass correctly 1`] = `"<a class="text-sm"></a>"`;

exports[`Link > renders with as correctly 1`] = `"<div type="button"></div>"`;

exports[`Link > renders with disabled correctly 1`] = `"<button type="button" disabled=""></button>"`;

exports[`Link > renders with inactiveClass correctly 1`] = `"<a class="text-gray-300"></a>"`;

exports[`Link > renders with to correctly 1`] = `"<a></a>"`;

exports[`Link > renders with type correctly 1`] = `"<button type="submit"></button>"`;
